xarray/core/dataset.py

```diff
@@ -1074,16 +1077,27 @@ def isel_points(self, dim='points', **indexers):
 
         indexer_dims = set(indexers)
 
+        def take(variable, slices):
+            if hasattr(variable.data, 'vindex'):
+                # Special case for dask backed arrays to use vectorised list indexing
+                sel = variable.data.vindex[slices]
+            else:
+                # Otherwise assume backend is numpy array with 'fancy' indexing
```

Comment:
Add a note to remove this helper function after numpy supports `vindex`, with a reference to https://github.com/numpy/numpy/pull/6075